networking.tf

```diff
@@ -0,0 +1,161 @@
+resource "aws_vpc" "vpc" {
+  cidr_block = local.cidr_block
+  tags = {
+    Name = var.name
+  }
+
+}
+
+data "aws_availability_zones" "available" {
+
+}
+
+
+resource "aws_subnet" "private" {
+  vpc_id            = aws_vpc.vpc.id
+  count             = local.subnets_count
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+  cidr_block        = local.private_subnets[count.index]
+
+  tags = {
+    Name = "private-subnet-${count.index}"
+  }
+}
+
+
+resource "aws_subnet" "public" {
+
+  vpc_id            = aws_vpc.vpc.id
+  cidr_block        = local.public_subnets[0]
+  availability_zone = data.aws_availability_zones.available.names[0]
+
+  tags = {
+
+    Name = "public-subnet"
+
+  }
+
+}
+
+resource "aws_route_table" "public" {
+
+  vpc_id = aws_vpc.vpc.id
+
+  tags = {
+    Name = "public-route-table"
+  }
+
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.vpc.id
+
+  tags = {
+    Name = "public-igw"
+  }
+}
+
+resource "aws_route" "public_internet_gateway" {
+
+  route_table_id         = aws_route_table.public.id
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = aws_internet_gateway.igw.id
+
+  timeouts {
+    create = "5m"
+  }
+}
+
+resource "aws_route_table_association" "public" {
+  subnet_id      = aws_subnet.public.id
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_eip" "nat" {
+
+  vpc = true
+
+  tags = {
+    Name = "aws-nat"
```

Comment:
use name prefix. Or else when we cannot use this module more than once, as the resources have same name.